satpy/readers/osisaf_l3_nc.py

```diff
@@ -0,0 +1,241 @@
+        from pyresample import create_area_def
+        try:
+            proj4str = self["Polar_Stereographic_Grid/attr/proj4_string"]
+        except KeyError:
+            # Some products don't have the proj str, so we construct it ourselves
+            sma = self["Polar_Stereographic_Grid/attr/semi_major_axis"]
+            smb = self["Polar_Stereographic_Grid/attr/semi_minor_axis"]
+            lon_0 = self["Polar_Stereographic_Grid/attr/straight_vertical_longitude_from_pole"]
+            lat_0 = self["Polar_Stereographic_Grid/attr/latitude_of_projection_origin"]
+            lat_ts = self["Polar_Stereographic_Grid/attr/standard_parallel"]
+            proj4str = f"+a={sma} +b={smb} +lat_ts={lat_ts} +lon_0={lon_0} +proj=stere +lat_0={lat_0}"
+        x_size = self["/dimension/xc"]
+        y_size = self["/dimension/yc"]
+        p_lowerleft_lat = self["lat"].values[y_size - 1, 0]
+        p_lowerleft_lon = self["lon"].values[y_size - 1, 0]
+        p_upperright_lat = self["lat"].values[0, x_size - 1]
+        p_upperright_lon = self["lon"].values[0, x_size - 1]
+        area_extent = [p_lowerleft_lon, p_lowerleft_lat, p_upperright_lon, p_upperright_lat]
+        area_def = create_area_def(area_id="osisaf_polar_stereographic",
+                                   description="osisaf_polar_stereographic",
+                                   proj_id="osisaf_polar_stereographic",
+                                   projection=proj4str, width=x_size, height=y_size, area_extent=area_extent,
+                                   units="deg")
+        return area_def
+
+
+    def _get_finfo_grid(self):
+        """Get grid in case of filename info being used."""
+        if self.filename_info["grid"] == "ease":
+            self.area_def = self._get_ease_grid()
+            return self.area_def
+        elif self.filename_info["grid"] == "polstere" or self.filename_info["grid"] == "stere":
+            self.area_def = self._get_polar_stereographic_grid()
+            return self.area_def
+        else:
+            raise ValueError(f"Unknown grid type: {self.filename_info['grid']}")
+
+    def _get_ftype_grid(self):
+        """Get grid in case of filetype info being used."""
+        if self.filetype_info["file_type"] == "osi_radflux_grid":
+            self.area_def = self._get_geographic_grid()
+            return self.area_def
+        elif self.filetype_info["file_type"] == "osi_sst":
+            self.area_def = self._get_polar_stereographic_grid()
+            return self.area_def
+
+    def get_area_def(self, area_id):
+        """Override abstract baseclass method"""
+        if "grid" in self.filename_info:
+            return self._get_finfo_grid()
+        else:
+            return self._get_ftype_grid()
+
+
+    def _get_ds_attr(self, a_name):
+        """Get a dataset attribute and check it's valid."""
+        try:
+            return self[a_name]
+        except KeyError:
+            return None
```

Comment:
Why swallow the `KeyError`s? Wouldn't that hide bugs elsewhere?

Reply:
If the main way this is used is to try multiple possible attribute/variable names, then maybe a different version of this with a for loop would be better?

```python
attr_val = self._get_first_attr(("/attr/X", "/attr/Y", "/attr/Z"))


def _get_first_attr(self, possible_names):
    for attr_name in possible_names:
        try:
            return self[attr_name]
        except KeyError:
            continue
    raise KeyError("...")

```

Reply:
Hopefully my newest change is suitable. If not, I'll come back to it some time in the new year.